HelpSource/Reference/MathJax_LaTeX_MathML.schelp

```diff
@@ -0,0 +1,321 @@
+TITLE::LaTeX and MathML Support via MathJax
+summary::How to use LaTeX Syntax in SCDocs using MathJax
+categories:: HelpSystem
+related:: Guides/WritingHelp, Classes/SCDoc, Classes/SCDocHTMLRenderer
+
+
+
+
+
+
+
+
+
+
+LaTeX and MathML syntax is available through MathJax scripts. The scripts are shipped with the SuperCollider installation and are automatically included by the link::https://www.mathjax.org/#gettingstarted#Getting Started:: (link::https://www.mathjax.org/#gettingstarted::)
+when rendering the SC-Doc to HTML. In principle, SuperCollider users do not have to deal with these scripts.
+
+
+
+
+
+
+
+
+
+
+
+section:: LaTeX support
+
+
+This document only covers the inline and display modes. For details of LaTeX syntax, see the LaTeX documentation:
+<link::https://www.latex-project.org/help/documentation/::>.
+
+
+
+subsection:: Inline mode
+
+
+Inline mode is used to insert a mathematical expression into the text flow and is achieved by enclosing the LaTeX code in a pair of code::\(:: and code::\):: delimiters. For example, the following LaTeX code will display the rendered mathematical expression \(a^2 + b^2 = c^2\) within a line of text:
+code::
+\(a^2 + b^2 = c^2\)
```

Comment:
Change all \( and \). The SCDoc syntax we agreed is math:: :: 

There is no reason to use the markdown syntax in a schelp file. math:: :: will be more consistent and ideomatic with this specific  markup

Reply:
I have here three questions again.

1. To confirm your proposal:
   Your proposal is to use SCDoc markdown syntax for LaTeX inline maths via MathJax.
   What is the corresponding part for LaTeX display maths via MathJax? Or do you think one of them is unnecessary? I think the advantage of implementing LaTeX via MathJax is the use of both inline and display maths. So I think there should be two SCDoc markdown syntaxes: 
    - `imath:: ... ::`: inline math
    - `dmath:: ... ::`: display math

2. There is currently an error when using `math:: ::`. It seems to need to be implemented by modifying some C++ source files. See the following paste of the relevant parts from the previous post: 
    >Have you tried using math:: ... ::? I cannot use it because sclang gives me the following error:
    >
    >ERROR: In C:\Temp\SuperCollider for Windows (x64)\HelpSource\Guides\WritingHelp_Punctuation_HTML_LaTeX_MathML.schelp:
 At line 1572: syntax error, unexpected ::, expecting end of file
To use SCDoc markup, I think we need to modify many parts of C++. Below is the relevant part from the parts I wrote earlier in this PR:
    >
    >You are absolutely right, but adding a new SCDoc modal tag is hard coded, and it seems to require adding a lexer in some cpp, hpp, l and y files in the source. See: https://github.com/supercollider/supercollider/issues/6131. You can see what I did to add the subsubsection tag in SCDoc and I had no success as I do not understand C++.

3. I cannot implement new SCDoc modal tags by modifying the sources written in C++. Who can implement MathJax in sclang?

4. After implementing `imath:: ... ::` and `dmath:: ... ::`, users can still use `\( ... \)` and `\[ ... \]`, and moreover the syntax provided by MathJax to bridge LaTeX is more convenient due to less typing markup. From the user's practice of using the LaTeX syntax, I would use `\( ... \)` and `\[ ... \]`. What do you think?

So I think we should change that part at least when we know how to implement `imath:: ... ::` and `dmath:: ... ::`.

Reply:
Yes, I can take a look and probably this part I will try to take care, or Mike. 

I was thinking about the difference between inline and displau.  Let me think about the best syntax. I'm not really convinced by dmath and imath, it can create confusion.

Reply:
@prko how code:: :: works both inline and display? or there is another token?

Reply:
```
The following tags (CODE and TELETYPE) can be written in two forms, either inline or block. Inline form has the end-tag on the same line, and need any enclosed double-colons that should be part of the text to be escaped with backslash (\). Block form has the tag and end-tag on lines by themselves, and can take multi-line text. In block form, only a single end-tag on its own line counts and any end-tags inside the text should not be escaped (except if it's on a single line).

code:: inline form ::

code::
block form :: <- needs no escaping
can span multiple lines
::
 CODE::
Render syntax-colored SuperCollider code.
 TELETYPE::
Render monospace text.
```

Reply:
In principal, math:: can handle both inline and display

Reply:
> The following tags (CODE and TELETYPE) can be written in two forms
> ... In principal, math:: can handle both inline and display

Correct, this is what I was trying to describe in the beginning. You would follow the implementation of `code::` to get both inline and display rendering from a single modal tag (`math::`).

> users can still use \( ... \) and \[ ... \]

Given that this is currently working I think we can just use that for the moment for the the functional prorotype, and given @prko has communicated that it will require more work/learning to implement the `math::` tag.

I think resolving the exact library specification is the priority to get buy in from others who are worried about the dependency.

Reply:
>In principal, math:: can handle both inline and display

aha!

- inline: `math::a^2::`
-  display:
   ```
   math::
   a^2
   ::
   ```

Thanks! It is very good!

Reply:
@smoge 
>@prko how code:: :: works both inline and display? or there is another token?
They are the same token, but are interpreted differently according to context!

Reply:
I removed `:teletype:`